src/test/FuzzerImpl.cpp

```diff
@@ -957,40 +957,216 @@ struct SponsoredEntryParameters
+        , mShortKey(shortKey)
+        , mNativeAssetAvailableForTestActivity(
+              nativeAssetAvailableForTestActivity)
+        , mOptionFlags(optionFlags)
     {
     }
 
-    int64_t const mAssetAvailableForTestActivity;
-    bool const mIssueAsset;
-};
+    constexpr AccountParameters(int shortKey,
+                                int64_t nativeAssetAvailableForTestActivity,
+                                uint32_t optionFlags, int sponsorKey)
+        : SponsoredEntryParameters(sponsorKey)
+        , mShortKey(shortKey)
+        , mNativeAssetAvailableForTestActivity(
+              nativeAssetAvailableForTestActivity)
+        , mOptionFlags(optionFlags)
+    {
+    }
 
-auto constexpr DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY = 256;
+    int const mShortKey;
+    int64_t const mNativeAssetAvailableForTestActivity;
+    uint32_t const mOptionFlags;
+};
 
 std::array<
     AccountParameters,
     FuzzUtils::NUMBER_OF_PREGENERATED_ACCOUNTS> constexpr accountParameters{
-    {{DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, true},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, true},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, true},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, true},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false},
-     {DEFAULT_ASSET_AVAILABLE_FOR_TEST_ACTIVITY, false}}};
+    {{0, 256, 0},
+     {1, 256, 0},
+     {2, 256, 0, 1}, // sponsored by account 1
+     {3, 256, AUTH_REQUIRED_FLAG},
+     {4, 256, 0},
+     {5, 256, 0},
+     {6, 256, 0},
+     {7, 256, 0},
+     {8, 256, 0},
+     {9, 256, 0},
+     {10, 256, 0},
+     {11, 256, 0},
+     {12, 256, 0},
+     {13, 256, 0},
+     {14, 256, 0},
+     {15, 256, 0}}};
+
+struct TrustLineParameters : public SponsoredEntryParameters
+{
+    constexpr TrustLineParameters(int trustor, AssetID const& assetID,
+                                  int64_t assetAvailableForTestActivity,
+                                  int64_t spareLimitAfterSetup)
+        : TrustLineParameters(trustor, assetID, false, 0,
+                              assetAvailableForTestActivity,
+                              spareLimitAfterSetup)
+    {
+        assert(!mAssetID.mIsNative);
+    }
+
+    static TrustLineParameters constexpr WithAllowTrust(
+        int trustor, AssetID const& assetID, uint32_t allowTrustFlags,
+        int64_t assetAvailableForTestActivity, int64_t spareLimitAfterSetup)
```

Comment:
Nitpick: parameter ordering. Why inject the extra parameters into the middle of the list? This will make it harder to look through a list of them because you'll have to remember the parameter ordering for every variation. Same applies for other variants.

Reply:
The idea is that the parameters of the shorter constructors are all in the same order as in the constructor that spells them all out explicitly:

```
    constexpr TrustLineParameters(int const trustor, AssetID const& assetID,
                                  bool callAllowTrustOp,
                                  uint32_t allowTrustFlags,
                                  int64_t assetAvailableForTestActivity,
                                  int64_t spareLimitAfterSetup, int sponsorKey)
```

So I'm looking at all the shorter constructors as convenience versions that _remove_ parameters from that list, not as some of the shorter constructors _adding_ parameters to other shorter constructors.  That feels more consistent to me, but as usual I can be convinced otherwise.  :)

Reply:
That's not compelling to me:
1. The constructor that you're using as the basis isn't public. It's not part of the public interface at all. You could rearrange all the parameters there and it wouldn't matter to anyone. In fact, users of this class explicitly should not care what that constructor looks like.
2. Given that you could rearrange all the parameters, why not put them in an order where the parameters that appear in every variant are at the beginning? Then you could satisfy your "remove" model and my "add" model simultaneously.

Reply:
> Given that you could rearrange all the parameters, why not put them in an order where the parameters that appear in every variant are at the beginning? Then you could satisfy your "remove" model and my "add" model simultaneously.

Ah, that I like a lot.  Done.